### PR TITLE
fix: session clear was not working correctly on CLI sessions

### PR DESCRIPTION
**Pull request by taka-oyama**
A better implementation of https://github.com/colopl/laravel-spanner/pull/46.

Has no beaking change so it can be applied to v4.

Implementation is a direct copy of https://github.com/googleapis/google-cloud-php/pull/5774.

The only exception are the lines added to `Connection`.

### CODE REVIEW

File: src/Session/DeleteOperation.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        $this->promises = $promises;
+    }
+
+    /**
+     * @return void
+     */
+    public function waitUntilComplete()
+    {
+        foreach ($this->promises as $promise) {
+            $promise->wait(false);
```
Comment (taka-oyama) [inline src/Session/DeleteOperation.php:43]: Disable unwrapping of rejected promises so it doesn't stop the code.

File: phpstan.neon [hunk 2]
```diff
@@ -27,3 +27,5 @@ parameters:
           path: src/Concerns/ManagesTransactions.php
         - message: '#^Parameter \#1 \$connection of method Illuminate\\Database\\DatabaseTransactionsManager::.+\(\) expects string, string\|null given\.$#'
           path: src/Concerns/ManagesTransactions.php
+        - message: "#.*#"
```
Comment (tyabu12) [inline phpstan.neon:30]: How about using excludePaths.analyse ?

https://phpstan.org/user-guide/ignoring-errors#excluding-whole-files

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,12 @@
 Added
 - Support Schema\Builder::getAllTables()
 
+Changed
+- Default SessionPool was changed from `Google\Cloud\Spanner\Session\CacheSessionPool` to `Colopl\Spanner\Session\CacheSessionPool` to patch an [unresolved issue on google's end](https://github.com/
```
Comment (tyabu12) [inline CHANGELOG.md:7]: it seems that the issue link is broken.

**Review by tyabu12 — CHANGES_REQUESTED** [review]

File: src/Session/DeleteOperation.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Copyright 2016 Google Inc.
```
Comment (halnique) [inline src/Session/DeleteOperation.php:3]: Is this correct?

**Review by halnique — COMMENTED** [review]
Does it not apply to the master branch?

File: phpstan.neon (hunk 2 above)
Comment (taka-oyama) [reply phpstan.neon]: Looks good, they have been added.

File: src/Session/DeleteOperation.php (hunk 4 above)
Comment (taka-oyama) [reply src/Session/DeleteOperation.php]: fixed.

**Review by halnique — APPROVED** [review]
LGTM

It would be a good idea to change the master branch (version 5.0 or later) to this approach as well.
Will this be changed in another PR?

**Comment by taka-oyama** [comment]
Yes, that's the plan. I think I will just remove the old fix from v5 branch.

**Pull request merged**